### PR TITLE
Bug 2087540: Adding popover to the CPU | Memory

### PR DESCRIPTION
**Pull request by DanaOrr**
Signed-off-by: Dana Orr <dorr@redhat.com>

## 📝 Description
Adding a popover to the CPU | Memory that shows how the calculation is performed.
In addition, each vm has its own calculation according to the amount of threads, cores and sockets it has.
 
 ## 🎥 Demo
CPU | Memory - VM
cpu-memory-vm

CPU | Memory - VMI
cpu-memory-vmi

CPU | Memory - Catalog 
cpu-memory-catalog

CPU | Memory - Template
cpu-memory-template

### CODE REVIEW

File: src/views/catalog/wizard/tabs/overview/components/WizardOverviewGrid.tsx [hunk 1]
```diff
@@ -128,6 +131,11 @@ const WizardOverviewGrid: React.FC<WizardOverviewGridProps> = ({ vm, tabsData, u
             title={t('CPU | Memory')}
             isEdit
             testId="wizard-overview-cpu-memory"
+            helperPopover={{
+              header: t('CPU | Memory'),
+              content: t(`CPUs = sockets X threads X cores.
+              This VM has ${sockets} sockets, ${threads} threads, and ${cores} cores. `),
```
Comment (upalatucci) [inline src/views/catalog/wizard/tabs/overview/components/WizardOverviewGrid.tsx:137]: For translations with variables, you should not use literals. 
Instead: 

```suggest
              content: t(`CPUs = sockets X threads X cores.
              This VM has {{sockets}} sockets, {{threads}} threads, and {{cores}}  cores.`, {
                sockets, 
                threads, 
                cores
              }),
``

File: src/views/virtualmachines/details/tabs/details/components/grid/leftGrid/VirtualMachineDetailsLeftGrid.tsx [hunk 2]
```diff
@@ -222,6 +225,9 @@ const VirtualMachineDetailsLeftGrid: React.FC<VirtualMachineDetailsLeftGridProps
           descriptionData={<CPUMemory vm={vm} />}
           descriptionHeader={t('CPU | Memory')}
           isEdit={canUpdateVM}
+          isPopover
+          bodyContent={` CPUs = sockets X threads X cores.
+           This VM has ${sockets} sockets, ${threads} threads, and ${cores} cores. `}
```
Comment (upalatucci) [inline src/views/virtualmachines/details/tabs/details/components/grid/leftGrid/VirtualMachineDetailsLeftGrid.tsx:230]: Make sure to translate also this string with `t('')`

File: src/views/catalog/wizard/tabs/overview/components/WizardOverviewGrid.tsx (hunk 1 above)
Comment (DanaOrr) [reply src/views/catalog/wizard/tabs/overview/components/WizardOverviewGrid.tsx]: Thank you Ugo!!

File: src/views/templates/details/tabs/details/components/CPUMemory.tsx [hunk 3]
```diff
@@ -50,7 +58,24 @@ const CPUMemory: React.FC<CPUMemoryProps> = ({ template }) => {
...
     <DescriptionListGroup>
-      <DescriptionListTerm>{t('CPU | Memory')}</DescriptionListTerm>
+      <DescriptionListTermHelpText>
+        <Popover
+          hasAutoWidth
+          maxWidth="30rem"
+          headerContent={t('CPU | Memory')}
+          bodyContent={t(
+            `CPUs = sockets X threads X cores.
+            This VM has {{sockets}} sockets, {{threads}} threads, and {{cores}}  cores.`,
```
Comment (avivtur) [inline src/views/templates/details/tabs/details/components/CPUMemory.tsx:68]: shouldn't we say here `This VM Template has...`?

File: src/views/catalog/wizard/tabs/overview/components/WizardOverviewGrid.tsx [hunk 4]
```diff
@@ -128,6 +131,18 @@ const WizardOverviewGrid: React.FC<WizardOverviewGridProps> = ({ vm, tabsData, u
             title={t('CPU | Memory')}
             isEdit
             testId="wizard-overview-cpu-memory"
+            helperPopover={{
+              header: t('CPU | Memory'),
+              content: t(
+                `CPUs = sockets X threads X cores.
+              This VM has {{sockets}} sockets, {{threads}} threads, and {{cores}}  cores.`,
```
Comment (avivtur) [inline src/views/catalog/wizard/tabs/overview/components/WizardOverviewGrid.tsx:138]: VM should be VirtualMachine

File: src/views/templates/details/tabs/details/components/CPUMemory.tsx [hunk 5]
```diff
@@ -50,7 +55,31 @@ const CPUMemory: React.FC<CPUMemoryProps> = ({ template }) => {
...
+            <>
+              <div>{t('CPUs = sockets X threads X cores.')} </div>
+              <div>
+                {t(
+                  'This VirtualMachine has {{sockets}} sockets, {{threads}} threads, and {{cores}}  cores.',
+                  {
+                    sockets: sockets,
+                    threads: threads,
+                    cores: cores,
+                  },
```
Comment (avivtur) [inline src/views/templates/details/tabs/details/components/CPUMemory.tsx:73]:
```suggestion
                  { sockets, threads, cores },
```

**Review by avivtur — COMMENTED** [review]
some more nits :)

File: src/views/catalog/wizard/tabs/overview/components/WizardOverviewGrid.tsx [hunk 6]
```diff
@@ -128,6 +129,20 @@ const WizardOverviewGrid: React.FC<WizardOverviewGridProps> = ({ vm, tabsData, u
...
             isEdit
             testId="wizard-overview-cpu-memory"
+            helperPopover={{
+              header: t('CPU | Memory'),
+              content: (
+                <>
+                  <div>{t('CPUs = sockets X threads X cores.')} </div>
+                  <div>
+                    {t(
+                      'This VirtualMachine has {{sockets}} sockets, {{threads}} threads, and {{cores}}  cores.',
```
Comment (avivtur) [inline src/views/catalog/wizard/tabs/overview/components/WizardOverviewGrid.tsx:139]: after {{cores} you have 2 spaces (' ') please remove one

File: src/views/templates/details/tabs/details/components/CPUMemory.tsx [hunk 7]
```diff
@@ -50,7 +55,26 @@ const CPUMemory: React.FC<CPUMemoryProps> = ({ template }) => {
...
+        <Popover
+          hasAutoWidth
+          maxWidth="30rem"
+          headerContent={t('CPU | Memory')}
+          bodyContent={
+            <>
+              <div>{t('CPUs = sockets X threads X cores.')} </div>
+              <div>
+                {t(
+                  'This VirtualMachine has {{sockets}} sockets, {{threads}} threads, and {{cores}}  cores.',
```
Comment (avivtur) [inline src/views/templates/details/tabs/details/components/CPUMemory.tsx:68]: here too

File: src/views/virtualmachines/details/tabs/details/components/grid/leftGrid/VirtualMachineDetailsLeftGrid.tsx [hunk 8]
```diff
@@ -222,6 +222,22 @@ const VirtualMachineDetailsLeftGrid: React.FC<VirtualMachineDetailsLeftGridProps
           descriptionData={<CPUMemory vm={vm} />}
           descriptionHeader={t('CPU | Memory')}
           isEdit={canUpdateVM}
+          isPopover
+          bodyContent={
+            <>
+              <div>{t('CPUs = sockets X threads X cores.')} </div>
+              <div>
+                {t(
+                  'This VirtualMachine has {{sockets}} sockets, {{threads}} threads, and {{cores}}  cores.',
```
Comment (avivtur) [inline src/views/virtualmachines/details/tabs/details/components/grid/leftGrid/VirtualMachineDetailsLeftGrid.tsx:231]: here too

File: src/views/virtualmachinesinstance/details/tabs/details/components/Details/CPUMemory/CPUMemory.tsx [hunk 9]
```diff
@@ -4,21 +4,50 @@ import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
...
+        <Popover
+          hasAutoWidth
+          maxWidth="30rem"
+          headerContent={t('CPU | Memory')}
+          bodyContent={
+            <>
+              <div>{t('CPUs = sockets X threads X cores.')} </div>
+              <div>
+                {t(
+                  'This VirtualMachine has {{sockets}} sockets, {{threads}} threads, and {{cores}}  cores.',
```
Comment (avivtur) [inline src/views/virtualmachinesinstance/details/tabs/details/components/Details/CPUMemory/CPUMemory.tsx:38]: here too

File: src/views/virtualmachinesinstance/details/tabs/details/components/Details/CPUMemory/CPUMemory.tsx [hunk 10]
```diff
@@ -1,24 +1,44 @@
...
+} from '@patternfly/react-core';
 
 type TolerationsProps = {
   vmi: V1VirtualMachineInstance;
 };
 
 const CPUMemory: React.FC<TolerationsProps> = ({ vmi }) => {
   const { t } = useKubevirtTranslation();
-
+  // const { cores, sockets, threads } = vmi?.spec?.domain?.cpu || {};
```
Comment (avivtur) [inline src/views/virtualmachinesinstance/details/tabs/details/components/Details/CPUMemory/CPUMemory.tsx:20]: can you please remove this?

**Review by avivtur — COMMENTED** [review]
looks a lot better :)

File: src/views/virtualmachinesinstance/details/tabs/details/components/Details/CPUMemory/CPUMemory.tsx (hunk 10 above)
Comment (DanaOrr) [reply src/views/virtualmachinesinstance/details/tabs/details/components/Details/CPUMemory/CPUMemory.tsx]: off course :)

**Pull request merged**